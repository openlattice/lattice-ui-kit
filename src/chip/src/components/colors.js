import * as Colors from '../../../colors';

const {
  NEUTRAL,
  PURPLE,
  BLUE,
  TEAL,
  GREEN,
  YELLOW,
  ORANGE,
  RED,
  MAGENTA,
  VIOLET
} = Colors;

export default {
  default: {
    background: NEUTRAL.N00,
    color: NEUTRAL.N600
  },
  blue: {
    background: BLUE.B00,
    color: BLUE.B400,
    clickable: BLUE.B100
  },
  green: {
    background: GREEN.G00,
    color: GREEN.G400,
    clickable: GREEN.G100
  },
  magenta: {
    background: MAGENTA.M00,
    color: MAGENTA.M400,
    clickable: MAGENTA.M100
  },
  orange: {
    background: ORANGE.O00,
    color: ORANGE.O500,
    clickable: ORANGE.O100
  },
  purple: {
    background: PURPLE.P00,
    color: PURPLE.P400,
    clickable: PURPLE.P100
  },
  red: {
    background: RED.R00,
    color: RED.R400,
    clickable: RED.R100
  },
  teal: {
    background: TEAL.T00,
    color: TEAL.T500,
    clickable: TEAL.T100
  },
  violet: {
    background: VIOLET.V00,
    color: VIOLET.V400,
    clickable: VIOLET.V100
  },
  yellow: {
    background: YELLOW.Y00,
    color: YELLOW.Y500,
    clickable: YELLOW.Y100
  },
  primary: {
    color: 'white'
  },
  secondary: {
    color: PURPLE.P400
  },
};
