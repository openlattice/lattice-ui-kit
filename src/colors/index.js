/*
 * @flow
 */

/*
 * Inspiration:
 * https://atlassian.design/guidelines/brand/color-1
 * https://bitbucket.org/atlassian/atlaskit-mk-2/src/master/packages/core/theme
 * https://evergreen.surge.sh/components/colors
 */

import {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
} from './src/Colors';

export {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
};

export default {
  BLUE,
  GREEN,
  MAGENTA,
  NEUTRAL,
  ORANGE,
  PURPLE,
  RED,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
};
