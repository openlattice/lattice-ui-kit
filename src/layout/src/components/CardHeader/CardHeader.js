import styled from 'styled-components';

import CardSegment from '../CardSegment/CardSegment';
import {
  GREEN_1,
  NEUTRALS,
  PURPLES,
  RED_1,
  WHITE,
  YELLOW_1,
} from '../../../../colors';
import { duration } from '../../../../style/transitions';
import { getStyleVariation } from '../../../../utils/StyleUtils';

const backgroundColor = getStyleVariation('mode', {
  danger: RED_1,
  default: NEUTRALS[0],
  primary: PURPLES[2],
  secondary: PURPLES[5],
  success: GREEN_1,
  warning: YELLOW_1
});

const getFontColor = getStyleVariation('mode', {
  danger: WHITE,
  default: WHITE,
  primary: WHITE,
  secondary: PURPLES[1],
  success: WHITE,
  warning: NEUTRALS[0],
});

const CardHeader = styled(CardSegment).attrs(() => ({
  as: 'header'
}))`
  align-items: baseline;
  background-color: ${backgroundColor};
  color: ${getFontColor};
  flex: none;
  transition: background-color ${duration.standard} ease-in-out,
    color ${duration.standard} ease-in-out;
`;

export default CardHeader;
