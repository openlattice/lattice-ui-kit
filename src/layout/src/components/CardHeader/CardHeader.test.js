import React from 'react';

import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import CardHeader from './CardHeader';

import * as Colors from '../../../../colors';

const {
  GREEN,
  NEUTRAL,
  PURPLE,
  RED,
  YELLOW,
} = Colors;

describe('CardHeader', () => {

  describe('snapshots', () => {
    test('without children', () => {
      const wrapper = mount(<CardHeader />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('with children', () => {
      const wrapper = mount(<CardHeader />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('render', () => {

    test('should render header', () => {
      const wrapper = mount(<CardHeader />);
      expect(wrapper.find('header')).toHaveLength(1);
    });

    describe('CardSegment inherited styles', () => {

      test('padding - sm', () => {
        const wrapper = mount(<CardHeader padding="sm" />);
        expect(wrapper).toHaveStyleRule('padding', '10px 30px 10px 30px');
      });

      test('padding - md', () => {
        const wrapper = mount(<CardHeader padding="md" />);
        expect(wrapper).toHaveStyleRule('padding', '20px 30px 20px 30px');
      });

      test('vertical', () => {
        const wrapper = mount(<CardHeader vertical />);
        expect(wrapper).toHaveStyleRule('flex-direction', 'column');
      });

    });

    describe('mode', () => {
      test('should render default mode', () => {
        const wrapper = mount(<CardHeader mode="default" />);
        expect(wrapper.prop('mode')).toEqual('default');
        expect(wrapper.find(CardHeader)).toHaveStyleRule('background-color', NEUTRAL.N100);
        expect(wrapper.find(CardHeader)).toHaveStyleRule('color', NEUTRAL.N900);
      });

      test('should render primary mode', () => {
        const wrapper = mount(<CardHeader mode="primary" />);
        expect(wrapper.prop('mode')).toEqual('primary');
        expect(wrapper.find(CardHeader)).toHaveStyleRule('background-color', PURPLE.P300);
        expect(wrapper.find(CardHeader)).toHaveStyleRule('color', 'white');
      });

      test('should render secondary mode', () => {
        const wrapper = mount(<CardHeader mode="secondary" />);
        expect(wrapper.prop('mode')).toEqual('secondary');
        expect(wrapper.find(CardHeader)).toHaveStyleRule('background-color', PURPLE.P100);
        expect(wrapper.find(CardHeader)).toHaveStyleRule('color', PURPLE.P400);
      });

      test('should render success mode', () => {
        const wrapper = mount(<CardHeader mode="success" />);
        expect(wrapper.prop('mode')).toEqual('success');
        expect(wrapper.find(CardHeader)).toHaveStyleRule('background-color', GREEN.G100);
        expect(wrapper.find(CardHeader)).toHaveStyleRule('color', GREEN.G400);
      });

      test('should render danger mode', () => {
        const wrapper = mount(<CardHeader mode="danger" />);
        expect(wrapper.prop('mode')).toEqual('danger');
        expect(wrapper.find(CardHeader)).toHaveStyleRule('background-color', RED.R100);
        expect(wrapper.find(CardHeader)).toHaveStyleRule('color', RED.R400);
      });

      test('should render warning mode', () => {
        const wrapper = mount(<CardHeader mode="warning" />);
        expect(wrapper.prop('mode')).toEqual('warning');
        expect(wrapper.find(CardHeader)).toHaveStyleRule('background-color', YELLOW.Y100);
        expect(wrapper.find(CardHeader)).toHaveStyleRule('color', YELLOW.Y500);
      });
    });

  });

  describe('props', () => {

    describe('onClick', () => {

      test('should have "cursor: inherit" for ":hover" by default', () => {
        const wrapper = mount(<CardHeader />);
        expect(wrapper).toHaveStyleRule('cursor', 'inherit', { modifier: ':hover' });
      });

      test('should have "cursor: pointer" for ":hover" when given onClick', () => {
        const wrapper = mount(<CardHeader onClick={() => { }} />);
        expect(wrapper).toHaveStyleRule('cursor', 'pointer', { modifier: ':hover' });
      });

    });

  });

});
