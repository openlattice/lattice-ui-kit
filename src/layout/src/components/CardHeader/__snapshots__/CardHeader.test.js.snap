// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader snapshots with children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  position: relative;
  border-bottom: 1px solid #dfdfe3;
  background-color: transparent;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 30px 30px 30px 30px;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-transition: background-color 200ms ease-out,color 200ms ease-out;
  transition: background-color 200ms ease-out,color 200ms ease-out;
}

.c0:hover {
  cursor: inherit;
}

.c0:first-child {
  border-radius: 3px 3px 0 0;
}

.c0:last-child {
  border-bottom: 0;
  border-radius: 0 0 3px 3px;
}

<CardHeader>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          [Function],
        ],
        "componentStyle": ComponentStyle {
          "componentId": "CardHeader-ez5xes-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "display:flex;flex:1 0 auto;position:relative;",
            [Function],
            " &:first-child{border-radius:3px 3px 0 0;}&:last-child{border-bottom:0;border-radius:0 0 3px 3px;}",
            "align-items:baseline;background-color:",
            [Function],
            ";color:",
            [Function],
            ";flex:none;transition:background-color ",
            "200ms",
            " ease-out,color ",
            "200ms",
            " ease-out;",
          ],
        },
        "displayName": "CardHeader",
        "foldedComponentIds": Array [
          "CardSegment-sc-44tqm2-0",
        ],
        "render": [Function],
        "styledComponentId": "CardHeader-ez5xes-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <header
      className="c0"
    />
  </StyledComponent>
</CardHeader>
`;

exports[`CardHeader snapshots without children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  position: relative;
  border-bottom: 1px solid #dfdfe3;
  background-color: transparent;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 30px 30px 30px 30px;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-transition: background-color 200ms ease-out,color 200ms ease-out;
  transition: background-color 200ms ease-out,color 200ms ease-out;
}

.c0:hover {
  cursor: inherit;
}

.c0:first-child {
  border-radius: 3px 3px 0 0;
}

.c0:last-child {
  border-bottom: 0;
  border-radius: 0 0 3px 3px;
}

<CardHeader>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          [Function],
        ],
        "componentStyle": ComponentStyle {
          "componentId": "CardHeader-ez5xes-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "display:flex;flex:1 0 auto;position:relative;",
            [Function],
            " &:first-child{border-radius:3px 3px 0 0;}&:last-child{border-bottom:0;border-radius:0 0 3px 3px;}",
            "align-items:baseline;background-color:",
            [Function],
            ";color:",
            [Function],
            ";flex:none;transition:background-color ",
            "200ms",
            " ease-out,color ",
            "200ms",
            " ease-out;",
          ],
        },
        "displayName": "CardHeader",
        "foldedComponentIds": Array [
          "CardSegment-sc-44tqm2-0",
        ],
        "render": [Function],
        "styledComponentId": "CardHeader-ez5xes-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <header
      className="c0"
    />
  </StyledComponent>
</CardHeader>
`;
